### PR TITLE
ENT-1858: Notary cleanup

### PR DESCRIPTION
**Pull request by adagys**
Migrated all non-BFT notary implementations to use async commits.

### CODE REVIEW

**Comment by adagys** [comment]
I've added a change to MockNetwork which needs a re-review

File: core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt [hunk 1]
```diff
@@ -34,14 +35,35 @@ abstract class NotaryServiceFlow(val otherSideSession: FlowSession, val service:
...
+            commitInputStates(parts, txId, requestPayload)
             signTransactionAndSendResponse(txId)
         } catch (e: NotaryInternalException) {
             throw NotaryException(e.error, txId)
         }
         return null
     }
 
+    @Suspendable
+    private fun commitInputStates(parts: TransactionParts, txId: SecureHash, requestPayload: NotarisationPayload) {
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt]: Why put this back into the shipped code in Corda?  If it remained in the CorDapp as it is now, we can change this outside of a full Corda release and/or not have to quite so carefully align the versions of the CorDapps with Corda releases.

e.g. if we fix that TODO, all the CorDapps need updating / become incompatible.  If we left it in the CorDapp, not so much...

File: core/src/main/kotlin/net/corda/core/internal/notary/UniquenessProvider.kt [hunk 2]
```diff
@@ -21,5 +21,13 @@ interface UniquenessProvider {
...
             references: List<StateRef> = emptyList()
-    )
+    ): CordaFuture<Result>
+
+    /** The outcome of committing a transaction. */
+    sealed class Result {
+        /** Indicates that all input states have been committed successfully. */
+        object Success : Result()
+        /** Indicates that the transaction has not been committed. */
+        data class Failure(val error: NotaryError) : Result()
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/internal/notary/UniquenessProvider.kt]: So now we don't throw an exception ?

File: core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt [hunk 3]
```diff
@@ -17,7 +18,7 @@ import net.corda.core.utilities.unwrap
  * Additional transaction validation logic can be added when implementing [validateRequest].
  */
 // See AbstractStateReplacementFlow.Acceptor for why it's Void?
-abstract class NotaryServiceFlow(val otherSideSession: FlowSession, val service: TrustedAuthorityNotaryService) : FlowLogic<Void?>() {
+abstract class NotaryServiceFlow(val otherSideSession: FlowSession, val service: SinglePartyNotaryService) : FlowLogic<Void?>() {
```
[the remark this answers is not in the document]
Comment (adagys) [reply core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt]: We might want to create an abstract BFT notary service class at some point. I would've called this one CFT, but it's used by both single-node and HA notaries.

File: core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt (hunk 1 above)
Comment (adagys) [reply core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt]: This logic already was in the core module, I just moved it from AsyncCFTNotaryService

File: core/src/main/kotlin/net/corda/core/internal/notary/UniquenessProvider.kt (hunk 2 above)
Comment (adagys) [reply core/src/main/kotlin/net/corda/core/internal/notary/UniquenessProvider.kt]: No, it gets a bit messy if we want to catch it through the async operation boundary

File: core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt (hunk 1 above)
Comment (adagys) [reply core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt]: Also note that the SimpleNotaryService is in a "virtual" CorDapp, so we'd always need Corda releases to update it

Comment (rick-r3) [reply core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt]: But the NotaryService impl. was in the CorDapp, so could be overridden.  Even though we generally just used one in the Corda codebase.  I'd rather we had maximum flexibility in the CorDapps.

Comment (adagys) [reply core/src/main/kotlin/net/corda/core/internal/notary/NotaryServiceFlow.kt]: Okay, fair, I'll move it to SinglePartyNotaryService

**Pull request merged**